org.testeditor.web.dropwizard/src/main/java/org/testeditor/web/dropwizard/BuildVersionProvider.xtend

```diff
@@ -0,0 +1,38 @@
+package org.testeditor.web.dropwizard
+
+import java.nio.charset.StandardCharsets
+import java.nio.file.Files
+import java.nio.file.Paths
+
+class BuildVersionProvider {
+		
+	var Iterable<String> dependencies
+	var Iterable<String> testeditorDependencies
+	var String dependencyId 
+	
+	private def void cacheDependencies(DropwizardApplicationConfiguration configuration, String optionalDependency) {
+		if (dependencies === null || configuration.applicationId != dependencyId) {
```

Comment:
if `optionalDependency` is provided, shouldn't we use that one to check against `dependencyId`?
Say, if `configuration.applicationId === 'foo.bar'`, and `dependencyId === 'foo.bar'` (cached from an earlier request), and now I request with `optionalDependency === 'x.y'`, the condition evaluates to `false`, meaning the dependencies of `foo.bar` will be returned, instead of the requested ones for `x.y`.
(Unless I'm missing something…)